python/cuml/metrics/cluster/adjustedrandindex.pyx

```diff
@@ -61,31 +61,28 @@ def adjusted_rand_score(labels_true, labels_pred, handle=None):
         if handle is None else handle
     cdef cumlHandle* handle_ =\
         <cumlHandle*><size_t>handle.getHandle()
-    if labels_true.astype != np.int64:
-        warnings.warn(" The dtype of ground truth is not int32"
-                      " converting the ground truth to int32")
-        labels_true = labels_true.astype(np.int32)
-    if labels_pred.astype != np.int32:
-        warnings.warn(" The dtype of predicted labels is not int32"
-                      " converting the predicted labels to int32")
-        labels_pred = labels_pred.astype(np.int32)
-
-    min_val_y = np.nanmin(labels_true)
-    lower_class_range = np.nanmin(labels_pred) \
-        if min_val_y > np.nanmin(labels_pred) else np.nanmin(labels_true)
-    max_val_y = np.nanmax(labels_true)
-    upper_class_range = np.nanmax(labels_pred) \
-        if max_val_y < np.nanmax(labels_pred) else np.nanmax(labels_true)
-    cdef uintptr_t y_ptr, y_hat_ptr
-    y_m, y_ptr, n_rows, _, _ = \
-        input_to_dev_array(labels_true)
-
-    y_hat_m, y_hat_ptr, _, _, y_hat_dtype = \
-        input_to_dev_array(labels_pred)
+
+    labels_true, n_rows, _, _ = \
+        input_to_cuml_array(labels_true, order='C', check_dtype=np.int32,
+                            convert_to_dtype=np.int32)
```

Comment:
We should add a convert_dtypeparameter to the function itself and then use it here (similar to the way the estimators do it), otherwise this will convert the data no matter whether the user wants it or not